tests/test_state.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+import time
+import rados
+from control.state import LocalGatewayState, OmapGatewayState, GatewayStateHandler
+
+
+@pytest.fixture(scope="module")
+def ioctx(config):
+    """Opens IO context to ceph pool."""
+    ceph_pool = config.get("ceph", "pool")
+    ceph_conf = config.get("ceph", "config_file")
+    conn = rados.Rados(conffile=ceph_conf)
+    conn.connect()
+    ioctx = conn.open_ioctx(ceph_pool)
+    yield ioctx
+    ioctx.close()
+
+
+@pytest.fixture
+def local_state():
+    """Returns local state object."""
+    return LocalGatewayState()
+
+
+@pytest.fixture
+def omap_state(config):
+    """Sets up and tears down OMAP state object."""
+    omap = OmapGatewayState(config)
+    omap.delete_state()
+    yield omap
+    omap.delete_state()
+
+
+def add_key(ioctx, key, value, version, omap_name, omap_version_key):
+    """Adds key to the specified OMAP and sets version number."""
+    with rados.WriteOpCtx() as write_op:
+        ioctx.set_omap(write_op, (key,), (value,))
+        ioctx.set_omap(write_op, (omap_version_key,), (str(version),))
+        ioctx.operate_write_op(write_op, omap_name)
+
+
+def remove_key(ioctx, key, version, omap_name, omap_version_key):
+    """Removes key from the specified OMAP."""
+    with rados.WriteOpCtx() as write_op:
+        ioctx.remove_omap_keys(write_op, (key,))
+        ioctx.set_omap(write_op, (omap_version_key,), (str(version),))
+        ioctx.operate_write_op(write_op, omap_name)
+
+
+def test_state_polling_update(config, ioctx, local_state, omap_state):
+    """Confirms periodic polling of the OMAP for updates."""
+
+    update_counter = 0
+
+    def _state_polling_update(update, is_add_req):
+        nonlocal update_counter
+        update_counter += 1
+        for k, v in update.items():
+            # Check for addition
+            if update_counter == 1:
+                assert is_add_req is True
+                assert k == key
+                assert v.decode("utf-8") == "add"
+            # Check for two-step change
+            if update_counter == 2:
+                assert is_add_req is False
+                assert k == key
+                assert v.decode("utf-8") == "changed"
+            if update_counter == 3:
+                assert is_add_req is True
+                assert k == key
+                assert v.decode("utf-8") == "changed"
+            # Check for removal
+            if update_counter == 4:
+                assert is_add_req is False
+                assert k == key
+            assert update_counter < 5
+
+    version = 1
+    update_interval_sec = 1
+    state = GatewayStateHandler(config, local_state, omap_state,
+                                _state_polling_update)
+    state.update_interval = update_interval_sec
+    state.use_notify = False
+    key = "bdev_test"
```

Comment:
I don't think `GatewayStateHandler` change had to lead to `bdev` being hard-coded here.  I'm disregarding this for now since it's test code, but I think we need to consolidate all on-disk format constants under a single entity and use them consistently in production and test.  It doesn't make sense to separate `OMAP_VERSION_KEY` (currently under `OmapGatewayState`) and e.g. `BDEV_PREFIX` (currently under `GatewayState`) when both are used in/define the same RADOS object.

Please send a follow up PR for that.